src/libmongoc/tests/test-mongoc-sdam-monitoring.c

```diff
@@ -859,6 +860,8 @@ _test_heartbeat_fails_dns (bool pooled)
       mongoc_client_destroy (client);
    }
 
+   duration = bson_get_monotonic_time () - start;
```

Comment:
In this case, you moved the `duration` calculation _after_ returning the client to the pool. Does this warrant a comment to balance out the earlier case of computing the `start` time _before_ pool popping?